src/main/java/spireMapOverhaul/zones/smithsFolly/events/RansackedForge.java

```diff
@@ -0,0 +1,162 @@
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.events.AbstractImageEvent;
+import com.megacrit.cardcrawl.helpers.CardLibrary;
+import com.megacrit.cardcrawl.localization.EventStrings;
+import com.megacrit.cardcrawl.potions.BlessingOfTheForge;
+import com.megacrit.cardcrawl.potions.PotionSlot;
+import com.megacrit.cardcrawl.rewards.RewardItem;
+import com.megacrit.cardcrawl.vfx.UpgradeShineEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndObtainEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardBrieflyEffect;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+import static spireMapOverhaul.SpireAnniversary6Mod.makeImagePath;
+
+public class RansackedForge extends AbstractImageEvent {
+    public static final String ID = makeID("RansackedForge");
+
+    private static final EventStrings eventStrings = CardCrawlGame.languagePack.getEventString(ID);
+
+    public static final String NAME = eventStrings.NAME;
+    public static final String[] DESCRIPTIONS = eventStrings.DESCRIPTIONS;
+    public static final String[] OPTIONS = eventStrings.OPTIONS;
+
+    private CurScreen screen = CurScreen.INTRO;
+    boolean hasStrike = AbstractDungeon.player.masterDeck.group.stream().anyMatch(card -> card.tags.contains(AbstractCard.CardTags.STARTER_STRIKE));
+
+    public RansackedForge() {
+        super(NAME, DESCRIPTIONS[0], makeImagePath("events/SmithsFolly/RansackedForge.png"));
+        this.screen = CurScreen.INTRO;
+
+        if (AbstractDungeon.player.masterDeck.findCardById(SearingBlow.ID) != null) {
+            this.imageEventText.setDialogOption(OPTIONS[0]);
+        } else if (AbstractDungeon.actNum == 3 && hasStrike) {
+            SearingBlow upgradedSearingBlow = new SearingBlow();
+            for (int i = 0; i < 4; i++) {
+                upgradedSearingBlow.upgrade();
+            }
+            this.imageEventText.setDialogOption(OPTIONS[1], upgradedSearingBlow);
+        } else {
+            SearingBlow upgradedSearingBlow = new SearingBlow();
+            for (int i = 0; i < 4; i++) {
+                upgradedSearingBlow.upgrade();
+            }
+            this.imageEventText.setDialogOption(OPTIONS[2], upgradedSearingBlow);
+        }
+
+        this.imageEventText.setDialogOption(OPTIONS[3]);
+        this.imageEventText.setDialogOption(OPTIONS[4]);
+    }
+
+
+    protected void buttonEffect(int buttonPressed) {
+        switch (screen) {
+            case INTRO:
+                switch (buttonPressed) {
+                    case 0:
+                        handleSearingBlowOption();
+                        break;
+                    case 1: // Gain potion slot and one Blessing of the Forge potion
+                        // Increase potion slots
+                        AbstractDungeon.player.potionSlots += 1;
+                        AbstractDungeon.player.potions.add(new PotionSlot(AbstractDungeon.player.potionSlots - 1));
```

Comment:
Does this work with save and reload? Just to be sure

Reply:
It does.